apps/namadillo/src/App/Common/TransactionReceipt.tsx

```diff
@@ -61,8 +67,20 @@ const TransferTransactionReceipt = ({
     );
   }, [transaction]);
 
+  const getEncodedViewingKey = (address: string): string => {
+    const canDecode = PseudoExtendedKey.can_decode(address);
+    if (!canDecode) return "";
```

Comment:
Maybe instead of checking `canDecode` twice you can just move this logic:
```
isExtendedKey(transaction.sourceAddress || "") ?
                  getEncodedViewingKey(transaction.sourceAddress || "")
                : transaction.sourceAddress
```
to a separate function?